src/System.CommandLine.Tests/TrimmingTestApp/TrimmingTestApp.csproj

```diff
@@ -0,0 +1,25 @@
+<Project Sdk="Microsoft.NET.Sdk">
+	
+	<PropertyGroup>
+		<OutputType>Exe</OutputType>
+		<TargetFramework>net6.0</TargetFramework>
+		<ImplicitUsings>enable</ImplicitUsings>
+		<Nullable>enable</Nullable>
+		<PublishTrimmed>true</PublishTrimmed>
+	</PropertyGroup>
+
+	<PropertyGroup>
+		<SystemCommandLineDllPath Condition="'$(SystemCommandLineDllPath)'==''">..\..\System.CommandLine\bin\Release\netstandard2.0\System.CommandLine.dll</SystemCommandLineDllPath>
```

Comment:
Why not just a ProjectReference?

Also - don't point to ns2.0 build. Point to net6.0.

Reply:
Since you aren't using a ProjectReference, all the warnings from System.CommandLine are getting collapsed into a single warning IL2014.

Here's the logic that says to not collapse warnings for ProjectReferences:

[https://github.com/dotnet/sdk/blob/8f1a762aee466715a984f5757a622b85daba5387/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ILLink.targets#L322-L325](https://github.com/dotnet/sdk/blob/8f1a762aee466715a984f5757a622b85daba5387/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ILLink.targets#L322-L325)

If you can't use a ProjectReference here, you can still get full warnings for this assembly by setting the `TrimmerSingleWarn=false` metadata on the System.CommandLine assembly.

Alternatively, since your app only uses System.CommandLine, it would be simpler to just turn off the single warning behavior globally:

```xml
<PropertyGroup>
  <TrimmerSingleWarn>false</TrimmerSingleWarn>
```

fyi @sbomer